lib/iris/util.py

```diff
@@ -1080,6 +1080,10 @@ def new_axis(src_cube, scalar_coord=None):
     if scalar_coord is not None:
         scalar_coord = src_cube.coord(scalar_coord)
 
+    if src_cube.shape is ():
+        if isinstance(src_cube.data, np.ma.core.MaskedConstant):
+            src_cube.data = np.ma.array(data=[0], mask=[1])
```

Comment:
Can we use `np.nan` as the data value here? We don;t know what it should be since a `MaskedConstant` has no value really, so not-a-number seems reasonable. I don't really like this part but I'm not sure if we have another choice...